pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1470,7 +1470,15 @@ func CheckNodeUnschedulablePredicate(pod *v1.Pod, meta algorithm.PredicateMetada
 		return false, []algorithm.PredicateFailureReason{ErrNodeUnknownCondition}, nil
 	}
 
-	if nodeInfo.Node().Spec.Unschedulable {
+	// If pod tolerate unschedulable taint, it's also tolerate node.Spec.Unschedulable.
+	unschedulable := !v1helper.TolerationsTolerateTaint(pod.Spec.Tolerations, &v1.Taint{
```

Comment:
the var name makes this hard to follow, as does combining info about the node and info about the pod into a single variable below. would something like this be clearer?

```go
podToleratesUnschedulable := v1helper.TolerationsTolerateTaint(...)`
if nodeInfo.Node().Spec.Unschedulable && !podToleratesUnschedulable {
  return false, []algorithm.PredicateFailureReason{ErrNodeUnschedulable}, nil
}
```

Reply:
done